### PR TITLE
Update Julia wrapper

### PR DESCRIPTION
**Pull request by findmyway**
- [x] update docs
- [x] rename OpenSpiel_jll -> OpenSpiel

### CODE REVIEW

**Comment by lanctot** [comment]
Hi @findmyway , this looks great. We have not had time to take a look yet -- does it mean we could avoid the download/install of the custom Julia install as we had setup now?

Also there seems to be a conflict; we had updated `build_and_run_tests.sh` last week. I guess it should be an easy merge, though. Can you take a look?

**Comment by findmyway** [comment]
I just merged the latest master branch.

> does it mean we could avoid the download/install of the custom Julia version as we have setup now?

We still need to download and install a Julia binary with v1.3 or above. But we do not need to install a specific CxxWrap.jl like I said [here](https://github.com/deepmind/open_spiel/issues/173#issuecomment-601830925) anymore 😉 .

Most changes in this PR are to update the dependency of CxxWrap.jl to v0.10. Another extra modification is that an extra `BUILD_WITH_PYTHON=ON` flag is added. This is handy when people only want to build the Julia wrapper(without python related dependencies). So now the whole story is:

- If a user is using Julia v1.3 on Linux, he/she only needs a usable version for experiments. Then just install this package with `]add OpenSpiel` in the Julia REPL. The package manager will download a precompiled version I registered [here](https://github.com/JuliaRegistries/General/tree/master/O/OpenSpiel).
- If a user wants to use it on Mac or needs the latest features. He/she needs to build the `libspieljl` locally by with the `install.sh` and `build_and_test.sh` scripts.

File: open_spiel/julia/wrapper/spieljl.cc [hunk 1]
```diff
@@ -170,14 +173,16 @@ JLCXX_MODULE define_julia_module(jlcxx::Module& mod) {
...
-          return vals;
-        });
+        // wrapped.method("values", [](WrappedT ps) {
+        //   std::vector<WrappedVal> vals;
+        //   vals.reserve(ps.size());
+        //   for (auto const& it : ps) {
+        //     vals.emplace_back(it.second);
+        //   }
+        //   return vals;
+        // });
```
Comment (jblespiau) [inline open_spiel/julia/wrapper/spieljl.cc:183]: Should this remain here or be deleted?

Comment (findmyway) [reply open_spiel/julia/wrapper/spieljl.cc]: Deleted. Thanks. It used to be a workaround to resolve https://github.com/JuliaInterop/CxxWrap.jl/issues/190

**Comment by lanctot** [comment]
Hi @findmyway , hmm.. the julia tests are failing when I import it:

```
WARNING: both OpenSpiel and Base export "step"; uses of it in module Main must be qualified
can play tic_tac_toe: Error During Test at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:8
  Got exception outside of a @test
  UndefVarError: step not defined
  Stacktrace:
   [1] evaluate_bots(::CxxWrap.StdLib.UniquePtrAllocated{State}, ::Array{MCTSBotAllocated,1}, ::Int64) at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/runtests.jl:31
   [2] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:14
   [3] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [4] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:9
   [5] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [6] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:4
   [7] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [8] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:3
   [9] include at ./boot.jl:328 [inlined]
   [10] include_relative(::Module, ::String) at ./loading.jl:1105
   [11] include(::Module, ::String) at ./Base.jl:31
   [12] include(::String) at ./client.jl:424
   [13] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/runtests.jl:40
   [14] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [15] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/runtests.jl:38
   [16] include at ./boot.jl:328 [inlined]
   [17] include_relative(::Module, ::String) at ./loading.jl:1105
   [18] include(::Module, ::String) at ./Base.jl:31
   [19] include(::String) at ./client.jl:424
   [20] top-level scope at none:6
   [21] eval(::Module, ::Any) at ./boot.jl:330
   [22] exec_options(::Base.JLOptions) at ./client.jl:263
   [23] _start() at ./client.jl:460
  
can play single player: Error During Test at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:18
  Got exception outside of a @test
  UndefVarError: step not defined
  Stacktrace:
   [1] evaluate_bots(::CxxWrap.StdLib.UniquePtrAllocated{State}, ::Array{MCTSBotAllocated,1}, ::Int64) at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/runtests.jl:31
   [2] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:23
   [3] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [4] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:19
   [5] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [6] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:4
   [7] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [8] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:3
   [9] include at ./boot.jl:328 [inlined]
   [10] include_relative(::Module, ::String) at ./loading.jl:1105
   [11] include(::Module, ::String) at ./Base.jl:31
   [12] include(::String) at ./client.jl:424
   [13] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/runtests.jl:40
   [14] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [15] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/runtests.jl:38
   [16] include at ./boot.jl:328 [inlined]
   [17] include_relative(::Module, ::String) at ./loading.jl:1105
   [18] include(::Module, ::String) at ./Base.jl:31
   [19] include(::String) at ./client.jl:424
   [20] top-level scope at none:6
   [21] eval(::Module, ::Any) at ./boot.jl:330
   [22] exec_options(::Base.JLOptions) at ./client.jl:263
   [23] _start() at ./client.jl:460
  
play three player stochastic games: Error During Test at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:27
  Got exception outside of a @test
  UndefVarError: step not defined
  Stacktrace:
   [1] evaluate_bots(::CxxWrap.StdLib.UniquePtrAllocated{State}, ::Array{MCTSBotAllocated,1}, ::Int64) at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/runtests.jl:31
   [2] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:34
   [3] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [4] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:28
   [5] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [6] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:4
   [7] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [8] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/bots.jl:3
   [9] include at ./boot.jl:328 [inlined]
   [10] include_relative(::Module, ::String) at ./loading.jl:1105
   [11] include(::Module, ::String) at ./Base.jl:31
   [12] include(::String) at ./client.jl:424
   [13] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/runtests.jl:40
   [14] top-level scope at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Test/src/Test.jl:1107
   [15] top-level scope at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/runtests.jl:38
   [16] include at ./boot.jl:328 [inlined]
   [17] include_relative(::Module, ::String) at ./loading.jl:1105
   [18] include(::Module, ::String) at ./Base.jl:31
   [19] include(::String) at ./client.jl:424
   [20] top-level scope at none:6
   [21] eval(::Module, ::Any) at ./boot.jl:330
   [22] exec_options(::Base.JLOptions) at ./client.jl:263
   [23] _start() at ./client.jl:460
  
Test Summary:                            |  Pass  Error  Total
OpenSpiel.jl                             | 25872      3  25875
  Games                                  |    27            27
  games simulation                       | 25803         25803
  bots                                   |    19      3     22
    MCTSBot                              |    19      3     22
      can play tic_tac_toe               |            1      1
      can play single player             |            1      1
      play three player stochastic games |            1      1
      solve draw                         |     9             9
      solve loss                         |     6             6
      solve win                          |     4             4
  cfr                                    |     8             8
  Trajectory                             |    15            15
ERROR: LoadError: Some tests did not pass: 25872 passed, 0 failed, 3 errored, 0 broken.
in expression starting at /usr/local/google/home/lanctot/tmp/open_spiel/open_spiel/julia/test/runtests.jl:37
ERROR: Package OpenSpiel errored during testing
Stacktrace:
 [1] pkgerror(::String, ::Vararg{String,N} where N) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/Types.jl:113
 [2] #test#131(::Bool, ::Nothing, ::Cmd, ::Cmd, ::typeof(Pkg.Operations.test), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/Operations.jl:1372
 [3] #test at ./none:0 [inlined]
 [4] #test#62(::Bool, ::Nothing, ::Cmd, ::Cmd, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Pkg.API.test), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:253
 [5] test(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:239
 [6] #test#58 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:233 [inlined]
 [7] test() at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:230
 [8] top-level scope at none:1


99% tests passed, 1 tests failed out of 84

Total Test time (real) =  84.58 sec

The following tests FAILED:
	 84 - julia_test (Failed)
Errors while running CTest
At least one test failed.
If this is the first time you have run these tests, try:
pip3 install -r requirements.txt
Note that outside a virtualenv, you will need to install the system 
wide matplotlib: sudo apt-get install python-matplotlib
Exiting virtualenv
lanctot@lanctot:open_spiel$ 
```

Any ideas?

**Comment by lanctot** [comment]
> Another edit, I missed this, which seems quite relevant!
> 
> ```
> WARNING: both OpenSpiel and Base export "step"; uses of it in module Main must be qualified
> ```
> 
> I guess we just have to tag on OpenSpiel:: , I'll try it

Prepending `step` with `OpenSpiel.` fixed it. I'm learning some Julia, cool!

**Comment by lanctot** [comment]
Ok, now the julia test seems to be failing on my Mac:

```
[100%] Building CXX object julia/CMakeFiles/spieljl.dir/wrapper/spieljl.cc.o

[100%] Linking CXX shared library libspieljl.dylib
[100%] Built target spieljl
Test project /Users/lanctot/open_spiel/build
    Start 146: julia_test
1/1 Test #146: julia_test .......................***Failed   22.73 sec
   Testing OpenSpiel
 Resolving package versions...
 Installed StatsBase ────── v0.33.0
 Installed DataStructures ─ v0.17.12
C++ exception while wrapping module OpenSpiel: Attempt to use unmapped type NSt3__16vectorIdNS_9allocatorIdEEEE in parameter list
ERROR: LoadError: Attempt to use unmapped type NSt3__16vectorIdNS_9allocatorIdEEEE in parameter list
Stacktrace:
 [1] register_julia_module at /Users/lanctot/.julia/packages/CxxWrap/LfTHV/src/CxxWrap.jl:406 [inlined]
 [2] readmodule(::String, ::Symbol, ::Module, ::Nothing) at /Users/lanctot/.julia/packages/CxxWrap/LfTHV/src/CxxWrap.jl:701
 [3] wrapmodule(::String, ::Symbol, ::Module, ::Nothing) at /Users/lanctot/.julia/packages/CxxWrap/LfTHV/src/CxxWrap.jl:705
 [4] top-level scope at /Users/lanctot/open_spiel/open_spiel/julia/src/OpenSpiel.jl:8
 [5] include at ./boot.jl:328 [inlined]
 [6] include_relative(::Module, ::String) at ./loading.jl:1105
 [7] include(::Module, ::String) at ./Base.jl:31
 [8] top-level scope at none:2
 [9] eval at ./boot.jl:330 [inlined]
 [10] eval(::Expr) at ./client.jl:425
 [11] top-level scope at ./none:3
in expression starting at /Users/lanctot/open_spiel/open_spiel/julia/src/OpenSpiel.jl:8
ERROR: LoadError: Failed to precompile OpenSpiel [7e5d8dc8-489f-4be6-9273-f4c3010e6ec1] to /Users/lanctot/.julia/compiled/v1.3/OpenSpiel/pEQx0_8bSIK.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1283
 [3] _require(::Base.PkgId) at ./loading.jl:1024
 [4] require(::Base.PkgId) at ./loading.jl:922
 [5] require(::Module, ::Symbol) at ./loading.jl:917
 [6] include at ./boot.jl:328 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1105
 [8] include(::Module, ::String) at ./Base.jl:31
 [9] include(::String) at ./client.jl:424
 [10] top-level scope at none:6
in expression starting at /Users/lanctot/open_spiel/open_spiel/julia/test/runtests.jl:1
ERROR: Package OpenSpiel errored during testing
Stacktrace:
 [1] pkgerror(::String, ::Vararg{String,N} where N) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/Types.jl:113
 [2] #test#131(::Bool, ::Nothing, ::Cmd, ::Cmd, ::typeof(Pkg.Operations.test), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/Operations.jl:1372
 [3] #test at ./none:0 [inlined]
 [4] #test#62(::Bool, ::Nothing, ::Cmd, ::Cmd, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Pkg.API.test), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:253
 [5] test(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:239
 [6] #test#58 at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:233 [inlined]
 [7] test() at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Pkg/src/API.jl:230
 [8] top-level scope at none:1


0% tests passed, 1 tests failed out of 1

Total Test time (real) =  22.87 sec

The following tests FAILED:
	146 - julia_test (Failed)
Errors while running CTest
```

**Comment by findmyway** [comment]
Ah, I can reproduce the error now.

It might be that I didn't test again after merging the latest master branch.

Will fix them soon.

**Comment by lanctot** [comment]
Great thanks, the Linux tests pass now so it' only the one error on MacOS.

**Comment by lanctot** [comment]
Hi @findmyway , I soled the first problem (conflict in step) in a different way, by prepending `OpenSpiel.` before `step` but I noticed you put back the imports in the header. Is there a preference between these?

Any idea on the MacOS problem..? Have you been able to reproduce it?

**Comment by findmyway** [comment]
> Is there a preference between these?

Yeah, I think import in the header would be better to avoid breaking things unexpectedly when using `step`, it also applies for `first` and `last`.

Unfortunately, I can't access my Mac environment recently. So it's not easy for me to debug the error with Mac. The error is quite strange to me. I didn't introduce any relevant change `spieljl.cc` file. And from the test result on Travis: https://travis-ci.org/github/findmyway/open_spiel/jobs/676503286 it seems to work fine. Could you check it again?

**Comment by lanctot** [comment]
Ok will change the step thing as you have.

The 146 tests is likely because I am only building a subset of the tests (I added the ability to test only specific tests) but I agree its odd to only be missing one.

I also upgraded my XCode and upgraded my Mac early this week which could explain it. 

Will try it again as soon as I can.

**Comment by lanctot** [comment]
Actually the 146 tests are because I didn't build with some of the optional deps (ACPC etc.)

I just tried again completely fresh and still get the same error. If it helps, it's coming from here: https://github.com/JuliaInterop/libcxxwrap-julia/blob/73149f8f1445d83dcb6ee2995e18b8f98f0c94e9/include/jlcxx/module.hpp#L343 . I wish it would give us a line number in `spieljl.cc` because I have no way of knowing where to look.

Since the tests are passing on Travis, it has to be my recent upgrades; but that is unsettling, because I'm sure others will hit this. I'm happy to merge it anyway but we should anticipate a bug report and/or maybe ask the CxxWrap devs.

**Comment by findmyway** [comment]
Thanks for trying it again!

```
C++ exception while wrapping module OpenSpiel: Attempt to use unmapped type NSt3__16vectorIdNS_9allocatorIdEEEE in parameter list
ERROR: LoadError: Attempt to use unmapped type NSt3__16vectorIdNS_9allocatorIdEEEE in parameter list
```

From my limited experience, it usually means some type is not wrapped with `jlcxx::stl::apply_stl`. (Sorry that I can't figure out the specific line without a workable environment.)

> Since the tests are passing on Travis, it has to be my recent upgrades (also didn't find anything on Google, likely because it's too new); but that is unsettling, because others will likely encounter this. I'm happy to merge it anyway but we should anticipate a bug report and/or maybe ask the CxxWrap devs.

Gently ping @barche to let him aware of this potential issue 😀 .

**Comment by lanctot** [comment]
> From my limited experience, it usually means some type is not wrapped with `jlcxx::stl::apply_stl`. (Sorry that I can't figure out the specific line without a workable environment.)

Yeah that's exactly what I was thinking too. I was trying to think if we've added anything recently (i.e. this week) that would cause this, and it doesn't seem so. So it must be some interaction between CxxWrap and the new XCode that maybe wasn't an issue before. 
 
> Gently ping @barche to let him aware of this potential issue 😀 .

Will do, thanks a lot for the help.

**Comment by barche** [comment]
I think the problem is that it is picking up a stale version of CxxWrap somehow, when compiling. After pulling in the code from this PR and running `build_and_run_tests.sh`, I found it was linked to the wrong lib:

```
ldd build/julia/libspieljl.so
libcxxwrap_julia.so.0 => /home/user/.julia/packages/CxxWrap/lDNAy/deps/usr/lib/libcxxwrap_julia.so.0
```

Deleting `/home/user/.julia/packages/CxxWrap/lDNAy` and `build/julia` and then building again fixed it for me.

Off-topic: I just happened to do these tests on our DGX station for reasons of convenience. Now that I have a working OpenSpiel installation, is there any cool stuff I can run to demo the power of our GPUs?

**Comment by lanctot** [comment]
Thanks @barche. Unfortunately, it didn't seem to fix my problem (which btw only appears on MacOS -- on linux it all works fine). I got rid of `$HOME/.julia` and `$HOME/.local/bin/julia*` and all of `build` to be sure, and re-ran the installer so it would re-install all the packages, and get the same problem when building.

**Comment by lanctot** [comment]
> Off-topic: I just happened to do these tests on our DGX station for reasons of convenience. Now that I have a working OpenSpiel installation, is there any cool stuff I can run to demo the power of our GPUs?

Ah! If you mean via the Julia API then I am not sure what is supported. Maybe Jun can say more.

There are several of the python algorithms that use Tensorflow ( but default to cpu ). I will check with the team to see if there is a better one for benchmarking and showing off hardware!

**Comment by barche** [comment]
OK, it took me a while to realize, but the problem is a recent incompatibility between the precompiled  libcxxwrap libraries and the latest Apple Clang, see https://github.com/JuliaInterop/CxxWrap.jl/issues/230

To fix this, compile libcxxwrap from source and set the Overrides.toml, as explained here: https://github.com/JuliaInterop/libcxxwrap-julia#building-libcxxwrap-julia

Unfortunately, I only realised this after building and testing the whole thing on my poor Macbook Air, it took 940 s for the tests, vs. 162 s on the DGX station ;)

**Comment by lanctot** [comment]
Ok, I've updated github now with these changes; I had imported the code prior to the last few commits which I added msyelf, so I'm closing the issue.

As for the outstanding issue with CxxWrap on MacOS with XCode 11.4: thanks @barche for all the time you've spent helping us understand this. Unfortunately, I have to move to other things now. I think it's ok to leave as is for now, though since I expect it would affect a very small portion of users for a short time, since I expect this will be eventually fixed once the prebuilt binaries in CxxWrap are updated. The other option would be to build `libcxxwrap` within the install of Julia, but that seems like a heavy change on our side for something that should correct itself soon. What do you think, @findmyway ? 

What I'm missing, though, is that there is no open issue on either CxxWrap or libcxxwrap documenting this as an outstanding "bug". Do the devs on their side know about it and is there a plan to fix it? @barche: are you one of the devs of these projects; could you comment on that? The issue you linked is closed so I'm not sure if my assumptions above are correct.

**Comment by findmyway** [comment]
> I think it's ok to leave as is for now, though, since I expect it would affect a very small portion of users for a short time, because this will be fixed once the prebuilt binaries in CxxWrap are updated.

Agree 👍

**Comment by barche** [comment]
> because this will be fixed once the prebuilt binaries in CxxWrap are updated

Unfortunately this is not possible: binaries for use with the "official" Julia releases are compiled using BinaryBuilder, which uses cross-compilation, so I expect this incompatibility with Xcode 11.4 to remain for a long time. To solve this for Julia end-users, OpenSpiel should be made available using BinaryBuilder as well, by submitting a build script to Yggdrasil: https://github.com/JuliaPackaging/Yggdrasil

**Comment by findmyway** [comment]
Thanks for @barche 's advice. In fact, I've already added one [there](https://github.com/JuliaPackaging/Yggdrasil/blob/master/O/OpenSpiel/build_tarballs.jl), but I encounter [some errors](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=2906&view=logs&jobId=28c390e1-87ba-505b-a007-392bba40f103&j=28c390e1-87ba-505b-a007-392bba40f103&t=f1dc5ea0-bf36-5c77-755f-69d10c63a694) when trying to build with `x86_64-apple-darwin14`. I'll revisit the problem sometime later.

**Comment by lanctot** [comment]
@findmyway, how about for now we just mention this problem in the Julia README.md (Under a 'known problems' section), just so people are aware?

**Pull request closed without merging**